src/diffusers/pipelines/versatile_diffusion/modeling_text_unet.py

```diff
@@ -759,6 +759,10 @@ def forward(
             if self.config.class_embed_type == "timestep":
                 class_labels = self.time_proj(class_labels)
 
+                # `Timesteps` does not contain any weights and will always return f32 tensors
+                # there might be better ways to encapsulate this.
+                class_labels = class_labels.to(dtype=self.dtype)
```

Comment:
```suggestion
                class_labels = class_labels.to(dtype=sample.dtype)
```

It's always safer IMO to use the dtype of other tensors that are necessarily defined before hand

Reply:
ah nice!